File: frontend/types/student-training.ts
// Mirrors backend/app/schemas/student_training.py, a read-only adapter
// over `industry_training` (database/migrations/023_industry_training.sql).

export interface StudentTrainingIndustry {
  id: string;
  company_name: string | null;
  industry_sector: string | null;
  logo_url: string | null;
}

export interface StudentTraining {
  id: string;
  title: string;
  description: string;
  location: string | null;
  work_mode: string | null;
  duration_months: number | null;
  capacity: number | null;
  eligibility_criteria: string | null;
  application_deadline: string | null;
  start_date: string | null;
  status: string;
  created_at: string | null;
  industry: StudentTrainingIndustry;
  has_applied: boolean;
}

export interface StudentTrainingListResponse {
  trainings: StudentTraining[];
}

File: backend/app/schemas/student_training.py
"""Pydantic schemas for the STUDENT side of Training discovery
(`industry_training`, database/migrations/023_industry_training.sql),
read-only. Mirrors app.schemas.student_workshop.
"""

from pydantic import BaseModel


class StudentTrainingIndustry(BaseModel):
    id: str
    company_name: str | None = None
    industry_sector: str | None = None
    logo_url: str | None = None


class StudentTraining(BaseModel):
    id: str
    title: str
    description: str
    location: str | None = None
    work_mode: str | None = None
    duration_months: int | None = None
    capacity: int | None = None
    eligibility_criteria: str | None = None
    application_deadline: str | None = None
    start_date: str | None = None
    status: str
    created_at: str | None = None
    industry: StudentTrainingIndustry
    has_applied: bool


class StudentTrainingListResponse(BaseModel):
    trainings: list[StudentTraining]
